src/superface-test.utils.ts

```diff
@@ -0,0 +1,243 @@
+import { NormalizedSuperJsonDocument } from '@superfaceai/ast';
+import {
+  ApiKeyPlacement,
+  isApiKeySecurityScheme,
+  isBasicAuthSecurityScheme,
+  isBearerTokenSecurityScheme,
+  isDigestSecurityScheme,
+  Profile,
+  Provider,
+  SecurityScheme,
+  SuperfaceClient,
+  SuperJson,
+  UnexpectedError,
+  UseCase,
+} from '@superfaceai/one-sdk';
+import { Definition as RecordingDefinition } from 'nock/types';
+import { join as joinPath } from 'path';
+import { URL } from 'url';
+
+import { CompleteSuperfaceTestConfig, SuperfaceTestConfigPayload } from '.';
+import {
+  ComponentUndefinedError,
+  InstanceMissingError,
+  SuperJsonNotFoundError,
+} from './common/errors';
+
+/**
+ * Asserts that entered sfConfig contains only instances of classes not strings.
+ */
+export function assertsPreparedConfig(
+  sfConfig: SuperfaceTestConfigPayload
+): asserts sfConfig is CompleteSuperfaceTestConfig {
+  assertsPreparedClient(sfConfig.client);
+  assertsPreparedProfile(sfConfig.profile);
+  assertsPreparedProvider(sfConfig.provider);
+  assertsPreparedUseCase(sfConfig.useCase);
+}
+
+export function assertsPreparedClient(
+  client: SuperfaceClient | undefined
+): asserts client is SuperfaceClient {
+  if (client === undefined) {
+    throw new ComponentUndefinedError('Client');
+  }
+}
+
+export function assertsPreparedProfile(
+  profile: Profile | string | undefined
+): asserts profile is Profile {
+  if (profile === undefined) {
+    throw new ComponentUndefinedError('Profile');
+  }
+
+  if (typeof profile === 'string') {
+    throw new InstanceMissingError('Profile');
+  }
+}
+
+export function assertsPreparedProvider(
+  provider: Provider | string | undefined
+): asserts provider is Provider {
+  if (provider === undefined) {
+    throw new ComponentUndefinedError('Provider');
+  }
+
+  if (typeof provider === 'string') {
+    throw new InstanceMissingError('Provider');
+  }
+}
+
+export function assertsPreparedUseCase(
+  useCase: UseCase | string | undefined
+): asserts useCase is UseCase {
+  if (useCase === undefined) {
+    throw new ComponentUndefinedError('UseCase');
+  }
+
+  if (typeof useCase === 'string') {
+    throw new InstanceMissingError('UseCase');
+  }
+}
+
+/**
+ * Checks whether provider is local and contains some file path.
+ */
+export function isProviderLocal(
+  provider: Provider | string,
+  profileId: string | undefined,
+  superJsonNormalized: NormalizedSuperJsonDocument
+): boolean {
+  const providerId = getProviderName(provider);
+
+  if (profileId !== undefined) {
+    const targetedProfileProvider =
+      superJsonNormalized.profiles[profileId].providers[providerId];
+
+    if (!('file' in targetedProfileProvider)) {
+      return false;
+    }
+  }
+
+  return true;
```

Comment:
Is this correct? If `profileId === undefined` this returns `true` for any provider, which doesn't seem correct to me. Shouldn't this require `profileId` to not be nullable?

Reply:
Yes, I'll fix that